[WIP] fix lint not failing in windows for bad template yamls

karuppiah7890 opened the pull request:
Fixes #6527 
Port of #6523 (v2 fix)
Twin of #6528 which was closed due to branch renaming

**What this PR does / why we need it**:
It fixes lint not failing in windows for bad template yamls

**If applicable**:
- [ ] this PR contains documentation
- [ ] this PR contains unit tests
- [ ] this PR has been tested for backwards compatibility

karuppiah7890 committed:
fix lint not failing in windows for bad template yamls

Signed-off-by: Karuppiah Natarajan <karuppiah7890@gmail.com>

thomastaylor312 commented:
@karuppiah7890 Just to make sure, this is marked as WIP because you are working on a unit test for it correct?

karuppiah7890 commented:
Correct! :)

On Wed, Oct 23, 2019, 11:09 PM Taylor Thomas <notifications@github.com>
wrote:

> @karuppiah7890 <https://github.com/karuppiah7890> Just to make sure, this
> is marked as WIP because you are working on a unit test for it correct?
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/helm/helm/pull/6757?email_source=notifications&email_token=ADBXB2BPMHJ4P6YXYXSRJNDQQCD6ZA5CNFSM4JD36NUKYY3PNVWWK3TUL52HS4DFVREXG43VMVBW63LNMVXHJKTDN5WW2ZLOORPWSZGOECCIN7I#issuecomment-545556221>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/ADBXB2HHZVYRXGBIAWTWODTQQCD6ZANCNFSM4JD36NUA>
> .
>

karuppiah7890 commented:
Similar to https://github.com/helm/helm/pull/6755#issuecomment-557781293 , I'm not able to build the code due to the presence of syscall.Umask in the code, which is not available in windows.

```
pkg/plugin/installer/http_installer_test.go
196:    oldmask := syscall.Umask(0000)
198:            syscall.Umask(oldmask)
```

thomastaylor312 commented:
@karuppiah7890 so that is only in a unit test. Is there any way we can replace that with something else or rewrite the test to not use it?

karuppiah7890 commented:
Yeah, I was checking about it. It's just that idk how permission masks make sense in Windows as it looks like it completely differently in Windows unlike Linux and Unix. I digged deeper but couldn't find much on a similar function for Windows. Also this has been quite tough and I just kept putting it away with every new hurdle 😅🙈 but this hurdle looks a bit harder. I'll see what can be done

thomastaylor312 commented:
Honestly I would shoot for rewriting the test so it tests the same functionality. Last resort we could consider putting this test in its own file with a build tag that excludes it on windows

karuppiah7890 commented:
I actually didn't know about the tags functionality. And yeah, I'll see about rewriting. 👍

bacongobbler commented:
ping. What's the current status on this PR?

karuppiah7890 commented:
I've paused work on this, since it has been quite a bit of work, with windows VMs. I'll get back to this soon, and when I do, I'll post here, but if someone else is picking it up, that's fine too 😄

Currently, I'm working on #6155

Commit:
fix(cmd): display warnings on stderr

The warnings introduced when a chart has been deprecated is displayed on standard out. This is a regression for users piping the output of `helm template` from a deprecated chart to `kubectl`. This changes the error message to display on standard error instead.

Signed-off-by: Matthew Fisher <matt.fisher@microsoft.com>

karuppiah7890 committed:
comment out syscall usage

Commit:
Fix golint issue

Signed-off-by: Guangwen Feng <fenggw-fnst@cn.fujitsu.com>

zouy414 committed:
Fix some go-lint warnings

Signed-off-by: zouyu <zouy.fnst@cn.fujitsu.com>

hickeyma committed:
Merge pull request #8366 from fenggw-fnst/work

Fix golint issue

pmengelbert committed:
Determine chart digest by manifest (#8249)

Currently, whenever the chart is printed, the digest of the .tar.gz
content layer is printed as the digest. The manifest digest is important
for OCI purposes, particularly in pushing to a registry.

Resolves #8248.

Signed-off-by: Peter Engelbert <pmengelbert@gmail.com>

Commit:
Merge pull request #8277 from bacongobbler/fix-deprecated-warning-on-stdout

fix(cmd): display warnings on stderr

hickeyma committed:
Merge pull request #8368 from Hellcatlk/go-lint

Fix some go-lint warnings

karuppiah7890 commented:
I'm closing this PR, because I don't have the bandwidth to attend to it.

Some info on this PR so that someone else can pick it up
* The fix is a very small fix - https://github.com/helm/helm/pull/6757/commits/851e7fc3145970163ede3cb0e1972b03f517a8f8
* The PR needed tests so that the bug doesn't pop up again. With a test the bug will be caught if it pops up.
* The testing can be done only in Windows OS as the thing we are testing is a bit OS specific - Windows OS using backslash `\` vs *nix OSes using forward slash `/`
* There is no CI/CD currently for running tests in Windows. Maintainers asked to still add tests that will catch the bug in Windows OS in case it pops up, and that running tests in Windows can be done later
* The codebase has lots of failing tests when running in Windows, so first those have to be fixed I think. And then we gotta check if the existing tests can help with verifying this issue's behavior or we can add an extra test. The failing tests are lot but the reasons for failure might be similar - like new line differences between OS and *nix systems that could be affecting assertions where we assert with test data (files) called [golden files](https://softwareengineering.stackexchange.com/questions/358786/what-is-golden-files)
* Even if the tests are fixed to run on Windows too, there's no guarantee that no one will break it again, given the CI/CD is present only for Linux, so not sure if it will be worth it to spend all the time in fixing the tests to make it work for Windows, and it can also be created as a separate issue and I would recommend adding Windows environment for testing for the CI/CD part if this is being done
